### ISSUE TITLE
Abacus Reporting is checking a field that doesn't exist

### ISSUE THREAD

**Issue by AzureEnigma** [cloudfoundry-incubator/cf-abacus#296]
Right now, in abacus-usage-reporting, there is a check in the code that attempts to get sw.quantity.current

```javascript
const setCurrentQuantity = (windows) => {
    map(windows, (w) => {
      map(w, (sw) => {
        if(sw)
          sw.quantity = sw.quantity.current;
      });
    });
  };
```

However, some of the aggregations do not have a quantity.current field and just use quantity directly. I was going to fix the check so that it also checks to see if the field exists before trying to use this.

```javascript
const setCurrentQuantity = (windows) => {
    map(windows, (w) => {
      map(w, (sw) => {
        if(sw && sw.quantity && sw.quantity.current)
          sw.quantity = sw.quantity.current;
      });
    });
  };
```

Is this the right way to do this? Thanks!

**Comment by betafood** [comment]
setCurrentQuantity is only called in terms of processing resource instance usage. Since resource instance usage is in the form of accumulated_usage, its quantity should follow the pattern of having a current and maybe a previous.

I don't believe valid input of this function would have a situation in which a current inside of quantity doesn't exist.

Assuming that this does happen though, then your solution looks like it would work out. Although, this is probably an edge case, if a resource defines a quantity that has 'current' as a property, it could get problematic.

**Comment by KRuelY** [comment]
It seems that the accumulate function returns null. Looking at the [code 206 - 239](https://github.com/cloudfoundry-incubator/cf-abacus/blob/master/lib/aggregation/accumulator/src/index.js#L206-L239), the only case that sw is equal to {} and quantity doesn't exist is when accumulated [returns null 215 - 218](https://github.com/cloudfoundry-incubator/cf-abacus/blob/master/lib/aggregation/accumulator/src/index.js#L215-L218).

**Comment by betafood** [comment]
The accum function at [lines 134 - 138](https://github.com/cloudfoundry-incubator/cf-abacus/blob/master/lib/aggregation/accumulator/src/index.js#L134-L138) instantiates all the windows as null (if no previous windows existed).

When it comes to [lines 215 - 218](https://github.com/cloudfoundry-incubator/cf-abacus/blob/master/lib/aggregation/accumulator/src/index.js#L215-L218), if the accumulate function returns null, then tw would still be null as nothing has been accumulated in that particular time window. If tw is defined, it means that it was accumulated as some point and thus should have the quantity.current structure.

**Comment by KRuelY** [comment]
@betafood is right, accumulator function guarantees that it is returning either null or { quantity }.

The problem is that aggregator [extends an empty object on null](https://github.com/cloudfoundry-incubator/cf-abacus/blob/master/lib/aggregation/aggregator/src/index.js#L362-L363). Instead of returning null, it is returning an empty object {}. 

Fixed in 9aca593f82b38535bc122888974772533b146781.
